src/client/Microsoft.Identity.Client/AppConfig/TokenRequestContentType.cs

```diff
@@ -12,11 +13,20 @@ namespace Microsoft.Identity.Client.AppConfig
     /// <summary>
     /// The media content type for post requests made to the identity provider.
     /// </summary>
-    public struct TokenRequestContentType
+    public class TokenRequestContentType
     {
+        internal string _value;
+
+        internal TokenRequestContentType(string value) { _value = value; }
+
         /// <summary>
         /// JSON media content type
         /// </summary>
-        public const string JSON = "application/json";
+        public static TokenRequestContentType JSON { get { return new TokenRequestContentType("application/json"); } }
```

Comment:
Avoid creating new objects (strings) where possible. This  a string constant that can be reused. 

In any case, an enum is much simpler to reason with here.